tests/ttnn/sweep_tests/shard_utility.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+
+import torch
+
+import ttnn
+
+
+def roundTo32(x):
+    if x < 32:
+        return 32
+    else:
+        if x % 32 == 0:
+            return x
+        else:
+            return ((x // 32) * 32) + 32
+
+
+def sharded_run(
+    dtype,
+    height,
+    width,
+    layout,
+    input_shard_orientation,
+    input_num_cores_x,
+    input_num_cores_y,
+    input_shard_strategy,
+    interleaved_to_sharded,
+    sharded_to_interleaved,
+    reshard,
+    device,
+    output_shard_orientation=None,
+    output_num_cores_x=None,
+    output_num_cores_y=None,
+    output_shard_strategy=None,
+):
+    input_num_cores_height = 1
+    input_num_cores_width = 1
+    if input_shard_strategy == ttnn.ShardStrategy.HEIGHT:
+        input_num_cores_height = input_num_cores_x * input_num_cores_y
+    elif input_shard_strategy == ttnn.ShardStrategy.WIDTH:
+        input_num_cores_width = input_num_cores_x * input_num_cores_y
+    elif input_shard_strategy == ttnn.ShardStrategy.BLOCK:
+        if input_shard_orientation == ttnn.ShardOrientation.ROW_MAJOR:
+            input_num_cores_height = input_num_cores_y
+            input_num_cores_width = input_num_cores_x
+        else:
+            input_num_cores_height = input_num_cores_x
+            input_num_cores_width = input_num_cores_y
+
```

Comment:
Can we set tensor height and width and sweep across arbitrary grids? This way we actually test partial shards. The way you have it will always result in nice full shards.

Reply:
Ya I was hoping to do that in a subsequent PR. The current sweep did full shards and it uncovered a couple hangs. 
I was hoping to add more coverage (Block sharding, uneven shards) in a subsequent PR.